Include/internal/pycore_abstract.h

```diff
@@ -8,6 +8,18 @@ extern "C" {
 #  error "this header requires Py_BUILD_CORE define"
 #endif
 
+/// Return 1 if the specified index is in the range [0, limit)
```

Comment:
I'm always confused by range syntax, so I prefer a simple:

```suggestion
// Return 1 if 0 <= index < limit
```